src/io/github/sspanak/tt9/ime/TraditionalT9.java

```diff
@@ -732,19 +732,20 @@ private boolean performOKAction() {
 				String oldText = textField.getTextBeforeCursor() + textField.getTextAfterCursor();
 
 				sendDownUpKeyEvents(KeyEvent.KEYCODE_DPAD_CENTER);
-
-				try {
-					// In Android there is no strictly defined confirmation key, hence DPAD_CENTER may have done nothing.
-					// If so, send an alternative key code as a final resort.
-					Thread.sleep(80);
-					String newText = textField.getTextBeforeCursor() + textField.getTextAfterCursor();
-					if (newText.equals(oldText)) {
-						sendDownUpKeyEvents(KeyEvent.KEYCODE_ENTER);
+				if (!oldText.isEmpty()) {
```

Comment:
Please revert the changes in this file. As I said, I have applied the change in AppHacks.